scripts/check_pre_commit_reqs.py

```diff
@@ -0,0 +1,97 @@
+    """Parse requirements file and return a dict of package versions."""
+    versions = {}
+    with open(filepath) as f:
+        for line in f:
+            line = line.strip()
+            if line and not line.startswith("#"):
+                # Handle different requirement formats
+                if "==" in line:
+                    pkg, version = line.split("==")
+                    versions[pkg.strip().lower()] = version.strip()
+    return versions
+
+
+def parse_precommit_config(filepath):
+    """Parse pre-commit config and extract ufmt repo rev and hook dependencies."""
+    with open(filepath) as f:
+        config = yaml.safe_load(f)
+
+    versions = {}
+    for repo in config["repos"]:
+        if "https://github.com/omnilib/ufmt" in repo.get("repo", ""):
+            # Get ufmt version from rev - assumes fixed format: vX.Y.Z
+            versions["ufmt"] = repo.get("rev", "").replace("v", "")
+
+            # Get dependency versions
+            for hook in repo["hooks"]:
+                if hook["id"] == "ufmt":
+                    for dep in hook.get("additional_dependencies", []):
+                        if "==" in dep:
+                            pkg, version = dep.split("==")
+                            versions[pkg.strip().lower()] = version.strip()
+            break
+    return versions
+
+
+def main():
+    # Find the pre-commit config and requirements files
+    config_file = Path(".pre-commit-config.yaml")
+    requirements_file = Path("requirements-fmt.txt")
+
+    if not config_file.exists():
+        print(f"Error: Could not find {config_file}")
+        sys.exit(1)
+
+    if not requirements_file.exists():
+        print(f"Error: Could not find {requirements_file}")
+        sys.exit(1)
+
+    # Parse both files
+    req_versions = parse_requirements(requirements_file)
+    config_versions = parse_precommit_config(config_file)
+
+    # Packages to check
+    packages = ["ufmt", "black", "usort", "ruff-api"]
+
+    # Check versions
+    mismatches = []
+    for pkg in packages:
+        req_ver = req_versions.get(pkg, None)
+        config_ver = config_versions.get(pkg, None)
+
+        if req_ver != config_ver:
+            found_version_str = f"{pkg}: {requirements_file} has {req_ver},"
+            if pkg == "ufmt":
+                mismatches.append(
+                    f"{found_version_str} pre-commit config rev has v{config_ver}"
+                )
+            else:
+                mismatches.append(
+                    f"{found_version_str} pre-commit config has {config_ver}"
+                )
+
+    # Report results
+    if mismatches:
+        print(
+            f"Version mismatches found:{''.join(f'\n\t{msg}' for msg in mismatches)}"
+            "\nPlease update the versions in `.pre-commit-config.yaml` to be "
+            "consistent with those in `requirements-fmt.txt` (source of truth)."
+            "\nNote: all versions must be pinned exactly ('==X.Y.Z') in both files."
```

Comment:
flake8 is complaining about this

Reply:
My solution would be to recommend turning off the rule based on the fact that backslashes are now allowed in python. I guess the issues is that in the monorepo you have internally there may be python versions at EOL where this is important. 

If locally changing the rule in the repo isn't a viable option because the config is global on the mirror then can look at ignoring the rule for this line?